bin/handler-slack-multichannel.rb

```diff
@@ -256,31 +263,44 @@ def post_data(notice, channel)
     http.use_ssl = true
 
     begin
-      req = Net::HTTP::Post.new("#{uri.path}?#{uri.query}")
-      if payload_template.nil?
-        text = slack_surround ? slack_surround + notice + slack_surround : notice
-        req.body = payload(text, channel).to_json
+      # Implement a retry strategy, with 5 tries max and a timeout of 10 seconds each
+      tries ||= 5
+      Timeout.timeout(10) do
```

Comment:
if we are gonna use timeout we should expose options to specify http level timeouts otherwise they will be obscured by a generic timeout message. See https://github.com/sensu-plugins/sensu-plugins-http/pull/123 for additional context.

Reply:
As for configurable options, we probably would like to have not only the timeout but also the number of retries, and the sleeping time in between retries, as configurable settings. Something like I proposed on https://github.com/sensu-plugins/sensu-plugins-slack/pull/83/files

On the other hand, I am not sure we do need to go as deep as specifying `dns`, `open` and `read` timeouts. Could we not just rescue these errors from `Net::`?